File: Data Structures and Algorithms/Sword refers to offer/C++/1.cpp
//1
class Solution {
public:
    bool Find(int target, vector<vector<int> > array) {
        bool isFound=false;
		for(int i=0;i<array.size();i++)
		{
			for(int j=0;j<array[i].size();j++)
				if(false==isFound&&target==array[i][j])
					isFound=true;
			
		} 
                 
                
    }
    
    return isFound;
};

//2
class Solution {
public:
    bool Find(int target, vector<vector<int> > array) {
       bool res=false;
	   int row=array.size();
	   int col=array[0].size();
	   for(int i=0,j=col-1;(i>=0&&i<row)&&(j>=0&&j<col);)
	   {
	   	if(target==array[i][j])
	   	{
	   		res=true;
	   		break;
	   	}
	   	else if(target<array[i][j])
	   		j--;
	   	else if(target>array[i][j])
	   		i++;
	   } 
	   return res;
	}
};

File: Data Structures and Algorithms/Sword refers to offer/Python/1.py
# -*- coding:utf-8 -*-
class Solution:
    # array 二维列表
    def Find(self, target, array):
        # write code here
        m,n=len(array),len(array[0])
        i=0
        j=n-1
        while (i<m and j>=0):
            if target==array[i][j]:
                return True
            elif target<array[i][j]:
                j-=1
            elif target>array[i][j]:
                i+=1
        return False
    
"""
# -*- coding:utf-8 -*-
class Solution:
    # array 二维列表
    def Find(self, target, array):
        # write code here
        m=len(array)
        n=len(array[0])
        for i in range(m):
            for j in range(n):
                if target==array[i][j]:
                    return True
        return False
"""
